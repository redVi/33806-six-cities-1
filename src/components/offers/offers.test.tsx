import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Offers from '@/components/offers/offers';

const renderer = new ShallowRenderer();

const items = [
  {id: 1, location: {latitude: 51.23, longitude: 6.80, zoom: 13}, title: 'Title', rating: 0, price: 320},
  {id: 2, location: {latitude: 51.24, longitude: 6.81, zoom: 13}, title: 'Title', rating: 3.4, price: 120},
];

const city = {
  name: `Amsterdam`,
  location: {
    latitude: 51.23,
    longitude: 51.24,
    zoom: 13
  }
}

describe(`Offers`, () => {
  it(`renders correctly`, () => {
    renderer.render(
      <Offers
        items={items}
        city={city}
        activeItem={items[0]}
        handleChangeOffersFilter={jest.fn()}
        setActiveItem={jest.fn()} />
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
