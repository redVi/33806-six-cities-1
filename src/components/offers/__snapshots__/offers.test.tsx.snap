// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers renders correctly 1`] = `
<div
  className="cities__places-container container"
>
  <section
    className="cities__places places"
  >
    <h2
      className="visually-hidden"
    >
      Places
    </h2>
    <b
      className="places__found"
    >
      2 places to stay in Amsterdam
    </b>
    <PlacesForm
      filterOptions={
        Array [
          Object {
            "text": "Popular",
            "value": "popular",
          },
          Object {
            "field": "ASC",
            "text": "Price: low to high",
            "value": "price",
          },
          Object {
            "field": "DESC",
            "text": "Price: high to low",
            "value": "price",
          },
          Object {
            "field": "DESC",
            "text": "Top rated first",
            "value": "rating",
          },
        ]
      }
      onChangeOffersFilter={[MockFunction]}
    />
    <PlacesList
      places={
        Array [
          Object {
            "id": 1,
            "location": Object {
              "latitude": 51.23,
              "longitude": 6.8,
              "zoom": 13,
            },
            "price": 320,
            "rating": 0,
            "title": "Title",
          },
          Object {
            "id": 2,
            "location": Object {
              "latitude": 51.24,
              "longitude": 6.81,
              "zoom": 13,
            },
            "price": 120,
            "rating": 3.4,
            "title": "Title",
          },
        ]
      }
      setActiveItem={[MockFunction]}
    />
  </section>
  <div
    className="cities__right-section"
  >
    <CityMap
      canZoomChange={true}
      hasSelectedItem={
        Object {
          "latitude": 51.23,
          "longitude": 6.8,
          "zoom": 13,
        }
      }
      items={
        Array [
          Object {
            "id": 1,
            "location": Object {
              "latitude": 51.23,
              "longitude": 6.8,
              "zoom": 13,
            },
            "price": 320,
            "rating": 0,
            "title": "Title",
          },
          Object {
            "id": 2,
            "location": Object {
              "latitude": 51.24,
              "longitude": 6.81,
              "zoom": 13,
            },
            "price": 120,
            "rating": 3.4,
            "title": "Title",
          },
        ]
      }
      location={
        Object {
          "latitude": 51.23,
          "longitude": 6.8,
          "zoom": 13,
        }
      }
    />
  </div>
</div>
`;
